internal/characters/tighnari/skill.go

```diff
@@ -0,0 +1,59 @@
+package tighnari
+
+import (
+	"github.com/genshinsim/gcsim/internal/frames"
+	"github.com/genshinsim/gcsim/pkg/core/action"
+	"github.com/genshinsim/gcsim/pkg/core/attributes"
+	"github.com/genshinsim/gcsim/pkg/core/combat"
+)
+
+var skillFrames []int
+
+const skillHitmark = 20
+
+func init() {
+	skillFrames = frames.InitAbilSlice(15)
+}
+
+func (c *char) Skill(p map[string]int) action.ActionInfo {
+	ai := combat.AttackInfo{
+		ActorIndex: c.Index,
+		Abil:       "Vijnana-Phala Mine",
+		AttackTag:  combat.AttackTagElementalArt,
+		ICDTag:     combat.ICDTagNone,
+		ICDGroup:   combat.ICDGroupDefault,
+		StrikeType: combat.StrikeTypeDefault,
+		Element:    attributes.Dendro,
+		Durability: 25,
+		Mult:       skill[c.TalentLvlSkill()],
+	}
+
+	c.Core.QueueAttack(
+		ai,
+		combat.NewCircleHit(c.Core.Combat.Player(), 2, false, combat.TargettableEnemy),
+		0,
+		skillHitmark,
+	)
+
+	var count float64 = 3
+	if c.Core.Rand.Float64() < 0.5 {
+		count++
+	}
+	c.Core.QueueParticle("tighnari", count, attributes.Dendro, skillHitmark+c.ParticleDelay)
+	c.SetCDWithDelay(action.ActionSkill, 12*60, 13)
+
+	c.AddStatus(vijnanasuffusionStatus, 12*60, false)
+	c.SetTag(wreatharrows, 3)
+
+	if c.Base.Cons >= 2 {
+		// TODO: c2 affets skill?
```

Comment:
c2 shouldn't affect the skill. c2 should be starting when the e field spawns/starts, but the e gadget (if it is like the other gadgets) snaps on skillRelease. so the c2 should start at skillRelease+travel, once you add in frames.